studio-docs/source/sharing-graphs.mdx

```diff
@@ -0,0 +1,691 @@
+
+* [This repo](https://github.com/apollographql/embeddable-explorer-demo-react) demonstrates a React app that embeds the Explorer and uses `postMessage`.
+
+    * See in particular [`setupEmbedRelay.ts`](https://github.com/apollographql/embeddable-explorer-demo-react/blob/main/src/setupEmbedRelay.ts), which defines view-layer-agnostic logic for communicating between the `iframe`, your own page, and your GraphQL server.
+
+    * You can also run this example on CodeSandbox:
+
+    <a href="https://codesandbox.io/s/github/apollographql/embeddable-explorer-demo-react/tree/main/?fontsize=14&hidenavigation=1&theme=dark">
+      <img alt="Edit embedded-explorer-demo" src="https://codesandbox.io/static/img/play-codesandbox.svg" />
+    </a>
+
+
+
+* The Explorer embedded in this article uses [this component](https://github.com/apollographql/gatsby-theme-apollo/blob/main/packages/gatsby-theme-apollo-docs/src/components/embeddable-explorer.js) from Apollo's Gatsby documentation theme.
+
+### `postMessage` setup
+
+> Before you proceed, make sure you have a [public variant](#making-a-variant-public) of your graph in Apollo Studio.
+
+This section walks you through the code to set up an embedded Explorer that uses `postMessage`. It borrows code from [this `setupEmbedRelay.ts` example](https://github.com/apollographql/embeddable-explorer-demo-react/blob/main/src/setupEmbedRelay.ts) and removes some optional portions for clarity.
+
+#### 1. Define Explorer constants
+
+Your webpage and the embedded Explorer use the `postMessage` API to send messages to each other. Those messages each have a name that indicates the _type_ of message being posted.
+
+Let's define constants for each of those message types, along with a constant for our `iframe`'s `src` URL:
+
+<MultiCodeBlock>
+
+```ts:title=setupEmbedRelay.ts
+// URL for any embedded Explorer iframe
+export const EMBEDDABLE_EXPLORER_URL =
+  "https://explorer.embed.apollographql.com";
+
+// Message types for queries and mutations
+const EXPLORER_QUERY_MUTATION_REQUEST = "ExplorerRequest";
+const EXPLORER_QUERY_MUTATION_RESPONSE = "ExplorerResponse";
+
+// Message types for subscriptions (not covered in this setup)
+const EXPLORER_SUBSCRIPTION_REQUEST = "ExplorerSubscriptionRequest";
+const EXPLORER_SUBSCRIPTION_RESPONSE = "ExplorerSubscriptionResponse";
+const SUBSCRIPTION_TERMINATION = "ExplorerSubscriptionTermination";
+```
+
+</MultiCodeBlock>
+
+#### 2. Set up communication
+
+Next, let's define a function that sets up communication between a webpage and the embedded Explorer's `iframe`. This function defines a callback that will execute whenever `window.postMessage` is called by the `iframe`.
+
+<MultiCodeBlock>
+
+```ts:title=setupEmbedRelay.ts
+export function setUpEmbedRelay() {
+
+  // Callback definition
+  const onPostMessageReceived = (
+    event: MessageEvent<{
+      name?: string;
+      operation?: string;
+      operationId?: string;
+      operationName?: string;
+      variables?: string;
```

Comment:
```suggestion
      variables?: Record<string, string>;
```

Lets just make this an object everywhere in the docs. I will add suggestions for all the places it is used